core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -46,9 +46,18 @@ private[spark] object SerDe {
     dis.readByte().toChar
   }
 
-  def readObject(dis: DataInputStream): Object = {
+  def readObject(dis: DataInputStream, typeName: String = ""): Object = {
```

Comment:
Hmm - do we need it though ? In `bytesToRow` you can just do the conversion if the typename is float ? Something like
```
val obj = SerDe.readObject(dis)
if (schemaTypeName == "Float") obj.asInstanceOf[Double].floatValue() else obj
```